### PR TITLE
FiBo-204 settings logout

### PR DESCRIPTION
**Pull request by ChrisEwert**
I added a button in the fragment that allows users to log out. When clicking the button, the user loses authorization and is redirected to the login page. I also added a test to the SettingsFragmentTest file that tests these features.

### CODE REVIEW

**Review by jens1o — APPROVED** [review]
10/10 would approve again

File: android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/settings/SettingsFragment.java [hunk 1]
```diff
@@ -33,25 +33,34 @@
...
-    private void initializeDeleteButton() {
+    private void initializeButtons() {
         deleteUserButton.setOnClickListener(e -> openDialog());
+        logoutButton.setOnClickListener(e -> logout());
+    }
+
+    private void logout() {
+        ApplicationState.getInstance(getContext()).clearAuthorization();
+        Intent i = new Intent(getActivity(), LoginActivity.class);
+        startActivity(i);
```
Comment (Jonas-Karl) [inline android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/settings/SettingsFragment.java:63]:
```suggestion
          ActivityUtils.swapActivity(requireActivity(),LoginActivity.class,false);       
```

I added a #swapActivity method in ActivityUtils. It can be used if you want to change activities just like in that case. The benefit of the method is that you wont forget to enable or disable the allowBack flag. In the previous case the user could logout and press the back button. After that he would be logged out (no tokens) but could access the settings page again.

When I implemented the #swapActivity, I set the currentActivity paramter class to AppCompatActivity. Inorder to make this change work, you have to change the class from AppCompatActivity to just an Activity, because getActivity() does not provide a AppCompatActivity. It provides a FragmentActivity. The use of Activity allows both Fragment and AppCompatActivivities

File: android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/settings/SettingsFragmentTest.java [hunk 2]
```diff
@@ -160,4 +163,32 @@ public void deleteUserAccessTokenInvalidTest() throws InterruptedException {
...
+
+        // Check that password login field is displayed
+        onView(withId(R.id.login_password))
+                .check(matches(isDisplayed()));
+
+        // Check that login button is displayed
+        onView(withId(R.id.login_button))
+                .check(matches(isDisplayed()));
+    }
+}
```
Comment (Jonas-Karl) [inline android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/settings/SettingsFragmentTest.java:194]: Here you could also add a Test if the use can navigate back to the application with pressing the backbutton. This test already exists in other Classes so it should be an hopefully easy copy and paste test

File: android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/settings/SettingsFragment.java [hunk 3]
```diff
@@ -61,11 +70,9 @@ private void openDialog() {
         builder.setTitle(R.string.delete_user_confirmation_title);
```
Comment (Jonas-Karl) [inline android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/settings/SettingsFragment.java:70]: builder.setCancelable(true); is the default value and therefor an unnecessary line

File: android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/settings/SettingsFragment.java [hunk 4]
```diff
@@ -84,7 +91,7 @@ private void deleteUserRequest() {
         };
 
         Response.ErrorListener onError = error -> {
-            if (error.networkResponse != null) {
+            if (null != error.networkResponse) {
                 switch (error.networkResponse.statusCode) {
                     case 401:
                         Log.e(TAG, "Unauthorized");
```
Comment (Jonas-Karl) [inline android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/settings/SettingsFragment.java:97]: adding a Log.e(TAG, "Unknown Error") entry to the default: branch would be useful for debugging purposses even though it is not part of this PR

File: android-app/app/src/main/res/values/themes.xml [hunk 5]
```diff
@@ -49,15 +49,15 @@
...
         <item name="android:paddingTop">18dp</item>
         <item name="android:paddingBottom">18dp</item>
         <item name="cornerRadius">10dp</item>
         <!-- TODO: find right background color for delete user button -->
         <item name="backgroundTint">@color/top_bar_placeholder</item>
-        <item name="android:textColor">@color/delete_red</item>
+        <item name="android:textColor">@color/mainTextColor</item>
         <item name="android:textAlignment">textStart</item>
         <item name="android:layout_marginStart">5dp</item>
         <item name="android:layout_marginEnd">5dp</item>
```
Comment (Jonas-Karl) [inline android-app/app/src/main/res/values/themes.xml:63]: Personally I dont like this style, buuuut it is exactly like the mockup. If I remember correctly I'm not the only one who does not really like the way it looks so I'm not sure whether we should ignore it and tackle this issue later on or whether we should do it correctly already at the beginning. 

Nonetheless we got a ticket soon to apply M3 Material Design. Maybe we could include in that ticket to update these button accordingly to the Material Design. Ideally it should be enough to add a style attribute like style="@style/Widget.Material3.Button....." and maybe some shapeAppearance.

Because of that I think we should not tackle this issue in this PR 👍

**Review by Jonas-Karl — CHANGES_REQUESTED** [review]
I found some artifacts of the old PR about deleting a user. I will request changes here and you can decide whether you want to do everything or just a part of my changes because not everything is part of this PR and extends it a little bit.

**Review by Jonas-Karl — APPROVED** [review]
